src/app/cases/creator/case-creator-submit.component.ts

```diff
@@ -44,6 +48,25 @@ export class CaseCreatorSubmitComponent implements OnInit {
     return (sanitizedEditForm: CaseEventData) => this.casesService.validateCase(this.jurisdictionId, this.caseTypeId, sanitizedEditForm);
   }
 
+  private isCreatingDraft(): boolean {
+    return !this.eventTrigger.case_id;
+  }
+
+  saveDraft(): (caseEventData: CaseEventData) => Observable<Draft> {
+    if (this.eventTrigger.can_save_draft) {
+      if (this.isCreatingDraft()) {
+        return (caseEventData: CaseEventData) => this.draftService.createDraft(this.jurisdictionId,
+          this.caseTypeId,
+          caseEventData);
+      } else {
+        return (caseEventData: CaseEventData) => this.draftService.updateDraft(this.jurisdictionId,
+          this.caseTypeId,
+          this.eventTrigger.case_id.slice(CaseResolver.DRAFT.length),
```

Comment:
it's not immediate to figure out what this line is doing. Can we create a method on evenTrigger with a meaningful name ?

Reply:
addressed in the next [PR](https://github.com/hmcts/ccd-case-management-web/pull/183)